### DISCUSSION TITLE
Can't run HyperHDR on RPI3

### DISCUSSION THREAD

**Discussion by corckie** [awawa-dev/HyperHDR#201] [Q&A] [answered]
Hi, 

I'm planning to switch from Hyperion to HDR and can't run it on my RPI3.

Hyperion got installed and working.

With HyperHDR I downloaded HyperHDR-17.0.0.0-Linux-armv7l.deb package and successfully installed it via SSH.

There was a confirmation that service is up and running and I should be able to access it from web browser.

Unfortunately this is not the case, I'm still getting "Can't reach this page" information.

I have also tried running hyperhdr service as root instead of pi using following commands:

sudo systemctl disable --now hyperhdr@pi
sudo systemctl enable --now hyperhdr@root

Still no luck.

Could it be because I'm using x64 version of RPI OS?

**Comment by awawa-dev** [comment, marked as answer]
Hi
You should not have installed Hyperion NG and HyperHDR at the same time: the behavior is unpredictable due to ports conflicts.
When your service is not started then run from the console ```/usr/bin/hyperhdr``` and analyze the output log.
And your suspicions about the system may be right: for x64 Raspbian you should use aarch64 ```HyperHDR-17.0.0.0-Linux-aarch64.deb``` package instead.

**Reply by corckie** [reply]
Hi, first of all I would like to thank you for your quick reply. 

I thought I need ARM Version specifically. 

Following your advice, I have installed aarch64.deb and now it is working.

Hyperion is not an issue, I have changed ports in prior and also disabled its service.

As of premade image, I'm using RPI for other packages (like NodeRed), so wanted to install additional package.

Thanks again!

**Comment by awawa-dev** [comment]
Fastest and safest way for testing is to burn x64 Raspberry Pi OS with HyperHDR included: ```SD-card-image-rpi234-aarch64.zip``` in the release section.

**Reply by corckie** [reply]
Also which software screen grabber for Windows10 would you recommend? I'm using HyperionScreenCap and everything works well, but wonder if you have any other preferred solution.

**Reply by awawa-dev** [reply]
HyperionScreenCap and HyperHDR are using DirectX11, although with a completely different approach. 
The main difference is HyperHDR can provide also HDR tone mapping to correct captured frames (DirectX11 doesn't correct it on its own) and there is no such solution for HyperionScreenCap.
If you are happy with HyperionScreenCap and have disabled HDR in Windows10 then you can stay with it. Next HyperHDR v18 will probably offer better network driver which should improve network saturation especially if one unit is using Wifi.

**Reply by corckie** [reply]
Just understood that Windowsapp is not only server, but also screen grabber. Just installed, set up "Software Screen Capture" (my screen shows correctly), but what next? Tried "Screen Capturing" in Advanced options, but share button points the screen towards Windows instance of HyperHDR instead of RPI instance.

**Reply by awawa-dev** [reply]
If you have correct video stream using HyperHDR directx11 software grabber in the live preview on Windows10, then you can share it with other HyperHDR unit using flatbuffers. Short guide how to configure it:
https://github.com/awawa-dev/HyperHDR/discussions/149

**Reply by awawa-dev** [reply]
You can find forwarder in the advanced->network tab.
If the forwarder is not visible on your setup then please switch to 'advanced' view because it may be hidden.

**Reply by corckie** [reply]
Got it working. Thank you.

HDR Tone mapping is the main reason I'm switching from Hyperion.

_Next HyperHDR v18 will probably offer better network driver which should improve network saturation especially if one unit is using Wifi._

Both my PC and RPI are wired, so I would appreciate more FPS. Huge advantage of HSC is that it can go up to 60Hz and beyond when capturing screen and it's very noticeable in my case (using 120Hz OLED with minimal input lag). Do you plan to add higher refresh rates?

**Reply by awawa-dev** [reply]
Limit of capturing rate in software screen grabber is rather artificial (it's rather interval of timer which captures a frame), can be unlock for higher values but it can increase danger of locking the game's resources.  And could also fully saturate the network or the receiver, even if you are using ethernet. The frames are not compressed. HyperionScreenCap is using VERY aggressive reduction of the frame size. If you want a equivalent in HyperHDR software grabber then you should reduce width size first in DX11 grabber.
Also some other tips: https://github.com/awawa-dev/HyperHDR/discussions/173

**Reply by awawa-dev** [reply]
That's the default HyperionScreenCap quality and I think it can't be changed: https://github.com/awawa-dev/HyperHDR/discussions/186

**Reply by corckie** [reply]
> Limit of capturing rate in software screen grabber is rather artificial (it's rather interval of timer which captures a frame), can be unlock for higher values but it can increase danger of locking the game's resources.

Fine, but I think that having enabled "expert" setting should allow it, maybe with some additional confirmation.

>And could also fully saturate the network or the receiver, even if you are using ethernet. The frames are not compressed. 

Same here, I think that if anyone uses HyperHDR they are rather aware users.

>HyperionScreenCap is using VERY aggressive reduction of the frame size.

You are allowed to choose image scaling factor. I find it surprisingly detailed with 2160p and 64 times scaling. And that's what I'm using - 64 scale and 60 Hz.

>If you want a equivalent in HyperHDR software grabber then you should reduce width size first in DX11 grabber. Also some other tips: #173

Tried reducing it, but as you mentioned, artificial limit won't let me go past 30 FPS...

**Reply by awawa-dev** [reply]
if you are using 64 scaling factor then you've got frame: 60x33 pixels...typical setup has even more LEDs than such video input. And it's causing a mess like in the previous linked article. That can't be simply detailed. You can set such width (max 60px) in HyperHDR dx11 to give them equal chances. Don't know if I change it yet because of possible side effects.

**Reply by corckie** [reply]
I know it might sound weird, but I think that it's more detailed than that. My setup is 103 LEDs horizontal and 58 LEDs vertical. Yet slightest color change is noticeable, for example hovering cursor over tray icons is well visible in LEDs. Some UI elements in games like health bars are basically perfectly transformed into LED backlight. Might not be perfect, but I have no issues here.

When it comes to framerate, things get much more noticeable. Using HyperHDR capture, I can't really see added benefit of higher resolution, but LED lag just kills my eyes. 30 seconds into any video and I can clearly see that LEDs do not sync with the picture and even if it's one or two frames, my side vision just can't stop seeing it - headache guaranteed. Gaming is also pain - I'm so used to 120 fps and 120Hz, that I can't stand anything less. Desync in this scenario is way too much to handle.

I can provide some videos with comparison if you need them.

**Comment by corckie** [comment]
@awawa-dev I think I have tried every possible setting, but HyperHDR screen grabber is always few frames behind the image. Could you please add 60fps target in the next release? And an option to go even lower with resolution?

**Comment by awawa-dev** [comment]
Did you lower width in the DX software grabber as I asked you before? What can you gain if we change the framerate from 30FPS to 60FPS? 17 milliseconds gain and much greater risk of TV image shuttering while playing? Maybe there is some other bottleneck in your ambient system that is limiting the performance?

**Reply by corckie** [reply]
Yes, I have it set to 196 pix wide and 30 fps.

Is there a bottleneck? Probably, but I don't think it's related to my system. HyperHDR driven by Hyperion Screen Capture works perfectly well when set to 60 fps.

17 ms might not be much, but there is great difference for an eye. Watching a movie or playing a game - if there is slight delay in LEDs response,  your eyesight gets tired in a moment. Even with slow paced material like Planet Earth you can see it easily when camera changes. Your brain is disoriented when seeing this desynchronization thus it leads to headache. I can record slow-mo video to see what the latency is.

**Reply by awawa-dev** [reply]
That's too much. Please set it to 60px...just as HyperionScreenCap with 64 scale does.

**Reply by awawa-dev** [reply]
With DX grabber set for 196 pix width, the frame that is needed to be send by HyperHDR is x9 times greater in size than HyperionScreenCap frame of scale 64. That can't be even compared.

**Comment by awawa-dev** [comment]
If you want to check out the result then please 
- fork HyperHDR
- enable in project settings github action for your new repository
- then I can tell you which thing you need to change (even using the browser) to have 60Hz. Github action will build the installers for you.  

But I have seriuos doubts it will help in your case.

**Reply by corckie** [reply]
@awawa-dev  wow that's really kind of you. I'll do it tomorrow as soon as possible. If it does not help, I will try to look deeper. Thanks.

**Reply by corckie** [reply]
@awawa-dev okay done, I have enabled github actions. Would appreciate your support what to do next.

**Reply by awawa-dev** [reply]
Browse for that file, click edit in the upper toolbox and change from 30 to 60.
![obraz](https://user-images.githubusercontent.com/69086569/152651754-1f32bf3f-f2ee-4da5-9043-e4436c8f05fb.png)
Then provide some name for commit and save changes. Check the action-tab current building process.

**Reply by corckie** [reply]
Great, thank you. Waiting for changes to appear.

**Reply by corckie** [reply]
Hi @awawa-dev , I have one newbie question. The action completed successfully, how can I download this new package?

**Reply by awawa-dev** [reply]
click on the Artifacts result in the action tab, download it, extract it & run the installer.

**Reply by corckie** [reply]
Hey @awawa-dev,
sorry for late response, I was on a vacation for two weeks.

Just got back and wanted to try it out.

My Actions tab is empty: 

![image](https://user-images.githubusercontent.com/2586929/154838028-7cf04941-47d4-4cbf-8b4d-4d419c28485c.png)

What would you suggest?
